lib/Twig/BaseNodeVisitor.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Twig_BaseNodeVisitor can be used to make node visitors compatible with Twig 1.x and 2.x.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+abstract class BaseNodeVisitor implements Twig_NodeVisitorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public function enterNode(Twig_NodeInterface $node, Twig_Environment $env)
+    {
```

Comment:
this should throw an exception for custom implementations of the interface saying it is not supported instead of triggering a fatal error on the following typehint